lib/http/node.js

```diff
@@ -123,8 +123,14 @@ function getProxyAgent(proxyOptions, parsedUrl, destination, mock) {
       return mockAgent;
     }
   }
-
-  const agentId = createAgentId(agentOptions.protocol, agentOptions.hostname, agentOptions.keepAlive);
+  let destHost;
+  try {
+    destHost = new URL(destination).hostname;
+  } catch (err) {
```

Comment:
can we add a test for both cases?

Reply:
Hello @sfc-gh-dprzybysz, I added an agent cache test. https://github.com/snowflakedb/snowflake-connector-nodejs/pull/946/files#diff-e251ed9eb74ad284dfcd0fcc9a3d2e34d6b18d4b5cba06f21f040beeb0573013R10. This testing includes parsing URL